.circleci/config.yml

```diff
@@ -11,7 +11,59 @@ executors:
       shell: bash.exe
 
 jobs:
-  build:
+  build-linux:
+    machine:
+      image: ubuntu-1604:201903-01
+    resource_class: 2xlarge
+    steps:
+      - checkout # check out the code in the project directory
+      - run: pyenv global 3.5.2
+      - run: sudo apt-get update -y
+      - run: sudo apt-get install -y libgflags-dev
+      - run: SKIP_FORMAT_BUCK_CHECKS=1 PRINT_PARALLAL_OUTPUTS=1 make J=32 all check -j32
+
+  build-linux-release:
+    machine:
+      image: ubuntu-1604:201903-01
+    resource_class: 2xlarge
+    steps:
+      - checkout # check out the code in the project directory
+      - run: make release -j32
+
+  build-linux-lite:
+    machine:
+      image: ubuntu-1604:201903-01 
+    resource_class: 2xlarge
+    steps:
+      - checkout # check out the code in the project directory
+      - run: pyenv global 3.5.2
+      - run: SKIP_FORMAT_BUCK_CHECKS=1 PRINT_PARALLAL_OUTPUTS=1 LITE=1 make J=32 all check -j32
```

Comment:
is `J=32` necessary when we have `-j32`?

Reply:
From reading the code, I didn't find one of the two would be set to the other if not specified. Can you point me which one I should use?

Reply:
# The default is to run one job per core (J=100%).
    # See "man parallel" for its "-j ..." option.
    J ?= 100%

Tests should automatically use all cores. Only -j should be needed to parallelize build.

Reply:
I missed this comment. Will do as a follow up.

Reply:
I thought about it more. Maybe it's good to use 32 threads rather than 16 based on number of cores. Some of the tests might be I/O bounded so using more parallelism might improve the throughput. Right now the running time is good so I would leave it there for now. We can adjust it later.